gradle/publish.gradle

```
@@ -0,0 +1,81 @@
+apply plugin: 'java'
+apply plugin: 'maven-publish'
+apply plugin: 'com.mapbox.sdkRegistry'
+apply from: file("${rootDir}/gradle/artifact-settings.gradle")
+
+afterEvaluate {
+    publishing {
+        publications {
+            release(MavenPublication) {
+                from components.release
+                groupId project.ext.mapboxArtifactGroupId
+                artifactId project.ext.mapboxArtifactId
+                version project.ext.versionName
+
+                artifact sourcesJar {
+                    classifier "sources"
+                }
+
+                artifact javadocJar {
+                    classifier "javadoc"
+                }
+
+                pom.withXml {
+                    final mainNode = asNode()
+                    mainNode.appendNode('name', project.ext.mapboxArtifactTitle)
+                    mainNode.appendNode('description', project.ext.mapboxArtifactTitle)
+                    mainNode.appendNode('url', project.ext.mapboxArtifactUrl)
+
+                    final licenseNode = mainNode.appendNode('licenses').appendNode('license')
+                    licenseNode.appendNode('name', project.ext.mapboxArtifactLicenseName)
+                    licenseNode.appendNode('url', project.ext.mapboxArtifactLicenseUrl)
+                    licenseNode.appendNode('distribution', "repo")
+
+                    final developerNode = mainNode.appendNode('developers').appendNode('developer')
+                    developerNode.appendNode('id', project.ext.mapboxDeveloperId)
+                    developerNode.appendNode('name', project.ext.mapboxDeveloperName)
+                    developerNode.appendNode('email', project.ext.mapboxDeveloperEmail)
+                    developerNode.appendNode('organization', project.ext.mapboxDeveloperOrganization)
+                    developerNode.appendNode('organizationUrl', project.ext.mapboxDeveloperOrganizationUrl)
+
+                    final scmNode = mainNode.appendNode("scm")
+                    scmNode.appendNode("connection", project.ext.mapboxArtifactScmUrl)
+                    scmNode.appendNode("developerConnection", project.ext.mapboxArtifactScmUrl)
+                    scmNode.appendNode("url", project.ext.mapboxArtifactUrl)
+                }
+            }
+            debug(MavenPublication) {
+                from components.debug
+                groupId project.ext.mapboxArtifactGroupId
+                artifactId project.ext.mapboxArtifactId
+                version project.ext.versionName
+
+                artifact sourcesJar {
+                    classifier "sources"
+                }
+
+                artifact javadocJar {
+                    classifier "javadoc"
+                }
+            }
+        }
+    }
+}
+
+def sdkNameMap = [:]
+sdkNameMap["services"] = "com-mapbox-mapboxsdk-mapbox-sdk-services"
```

```suggestion
sdkNameMap["services"] = "mobile-mapbox-java-mapbox-sdk-services"
```

and same everywhere below

---

any reason why it needs to be in mobile-mapbox-java-..?

---

No, more for consistency with what's already up in the SDK Registry. I'm open to other suggestions.

---

`com-mapbox-mapboxsdk-mapbox-sdk-services` is the current location for the jcenter script migrated dependencies. While the root folder name can be selected arbitrarily, with the script creation I had to construct something unique that was derived from the jcenter data, which was the package name. My preference is to keep all versions in 1 place but I'm :+1: with moving the older versions to the proposed folder as well.

---

Ah, I forgot those are already in there. In this case, let's stick to `com-mapbox-mapboxsdk-*` :+1: